src/node_launcher.star

```diff
@@ -16,65 +16,72 @@ PUBLIC_IP = "127.0.0.1"
+
+def launch(plan, genesis, image, node_count, ephemeral_ports, min_cpu, min_memory, vmId, dont_start_subnets,
+           custom_subnet_vm_path, custom_subnet_vm_url):
     bootstrap_ips = []
     bootstrap_ids = []
     nodes = []
     launch_commands = []
 
     services = {}
     plan.print("Creating all the avalanche containers paralllely")
-    for index in range (0, node_count):
+    for index in range(0, node_count):
         node_name = NODE_NAME_PREFIX + str(index)
 
-        node_data_dirpath =  ABS_DATA_DIRPATH + node_name + "/"
+        node_data_dirpath = ABS_DATA_DIRPATH + node_name + "/"
         node_config_filepath = node_data_dirpath + "config.json"
 
         launch_node_cmd = [
             "nohup",
             "/avalanchego/build/" + EXECUTABLE_PATH,
-            "--genesis=/tmp/data/genesis.json", 
+            "--genesis-file=/tmp/data/genesis.json",
             "--data-dir=" + node_data_dirpath,
             "--config-file=" + node_config_filepath,
             "--http-host=0.0.0.0",
             "--staking-port=" + str(STAKING_PORT_NUM),
-            "--http-port="+ str(RPC_PORT_NUM),
-            "--log-dir=/tmp/"
+            "--http-port=" + str(RPC_PORT_NUM),
+            "--log-dir=/tmp/",
+            "--network-health-min-conn-peers=" + str(node_count - 1),
         ]
 
+        plan.print("Creating node {0} with command {1}".format(node_name, launch_node_cmd))
+
         public_ports = {}
         if not ephemeral_ports:
-            public_ports["rpc"] = PortSpec(number = RPC_PORT_NUM+ index*2 , transport_protocol = "TCP", wait=None)
-            public_ports["staking"] = PortSpec(number = STAKING_PORT_NUM + index*2 , transport_protocol = "TCP", wait=None)
+            public_ports["rpc"] = PortSpec(number=RPC_PORT_NUM + index * 2, transport_protocol="TCP", wait=None)
+            public_ports["staking"] = PortSpec(number=STAKING_PORT_NUM + index * 2, transport_protocol="TCP", wait=None)
 
         log_files = ["main.log", "C.log", "X.log", "P.log", "vm-factory.log"]
         log_files_cmds = ["touch /tmp/{0}".format(log_file) for log_file in log_files]
         log_file_cmd = " && ".join(log_files_cmds)
 
+        subnet_evm_plugin = plan.upload_files(
+            "../static_files/subnet-evm")
+
         node_service_config = ServiceConfig(
             image = image,
             ports = {
                 "rpc": PortSpec(number = RPC_PORT_NUM, transport_protocol = "TCP", wait = None),
                 "staking": PortSpec(number = STAKING_PORT_NUM, transport_protocol = "TCP", wait = None)
             },
-            entrypoint = ["/bin/sh", "-c", log_file_cmd + " && cd /tmp && tail -F *.log"],
-            files = {
+            entrypoint=["/bin/sh", "-c", log_file_cmd + " && cd /tmp && tail -F *.log"],
+            files={
                 "/tmp/data": genesis,
+                "/uploaded_plugins/": subnet_evm_plugin,
```

Comment:
Uploading the subnet-evm binary to later copy it to the ava plugin location. Directly loading it into ava plugin location caused an error because it needs renaming to include the VM name.